extras/kms/examples/docker-compose.yml

```diff
@@ -0,0 +1,34 @@
+version: "3.8"
+
+services:
+
+  vault-server:
+    image: vault:latest
+    ports:
+      - "8200:8200"
+    environment:
+      VAULT_ADDR: "http://0.0.0.0:8200"
+      VAULT_DEV_ROOT_TOKEN_ID: "vault-plaintext-root-token"
+    cap_add:
+      - IPC_LOCK
+    networks:
+      vault-network:
+        ipv4_address: 172.21.0.10
+        aliases:
+          - vault-server
+
+  vault-client:
+    build: .
+    environment:
+      VAULT_ADDR: "http://vault-server:8200"
+    networks:
+      vault-network:
+        ipv4_address: 172.21.0.20
+        aliases:
+          - vault-client
+
+networks:
+  vault-network:
+    ipam:
+      config:
+        - subnet: 172.21.0.0/24
```

Comment:
I don't think this network configuration is necessary, was there a reason for including it?

Reply:
the vault-client and vault-server both reference it to chat about bit.

Reply:
This works without any of the network configuration, docker-compose will create a network that both containers are a part of and they can communicate. The example still works with this diff applied:

```diff
diff --git a/extras/kms/examples/docker-compose.yml b/extras/kms/examples/docker-compose.yml
index 5c8e1bb..4da81ff 100644
--- a/extras/kms/examples/docker-compose.yml
+++ b/extras/kms/examples/docker-compose.yml
@@ -11,24 +11,8 @@ services:
       VAULT_DEV_ROOT_TOKEN_ID: "vault-plaintext-root-token"
     cap_add:
       - IPC_LOCK
-    networks:
-      vault-network:
-        ipv4_address: 172.21.0.10
-        aliases:
-          - vault-server

   vault-client:
     build: .
     environment:
       VAULT_ADDR: "http://vault-server:8200"
-    networks:
-      vault-network:
-        ipv4_address: 172.21.0.20
-        aliases:
-          - vault-client
-
-networks:
-  vault-network:
-    ipam:
-      config:
-        - subnet: 172.21.0.0/24
```

Reply:
TIL!  ty!